src/Symfony/Component/DependencyInjection/Compiler/ResolveDefinitionTemplatesPass.php

```diff
@@ -96,11 +97,12 @@ private function resolveArguments(ContainerBuilder $container, array $arguments,
      */
     private function resolveDefinition(ContainerBuilder $container, DefinitionDecorator $definition)
     {
-        if (!$container->hasDefinition($parent = $definition->getParent())) {
+        try {
+            $parentDef = $container->findDefinition($parent = $definition->getParent());
+        } catch (ServiceNotFoundException $e) {
             throw new RuntimeException(sprintf('The parent definition "%s" defined for definition "%s" does not exist.', $parent, $this->currentId));
```

Comment:
please link the exception as a previous exception, as it might give more details in case of a chain of aliases (telling which alias of the chain failed to resolve)